flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperator.java

```diff
@@ -413,8 +413,11 @@ public final OperatorSnapshotFutures snapshotState(long checkpointId, long times
 				snapshotException.addSuppressed(e);
 			}
 
-			throw new Exception("Could not complete snapshot " + checkpointId + " for operator " +
-				getOperatorName() + '.', snapshotException);
+			String snapshotFailMessage = "Could not complete snapshot " + checkpointId + " for operator " +
+				getOperatorName() + ".";
+
+			LOG.info(snapshotFailMessage, snapshotException);
```

Comment:
I think it would be better to log the failure in `RpcCheckpointResponder#declineCheckpoint` because it is there where the message leaves the `TaskExecutor` and we no longer have control over it. Moreover, we would cover other failures coming from the other calling paths as well.